articles/views.py

```diff
@@ -1,15 +1,49 @@
+from django.shortcuts import render, redirect
+from django.contrib.auth.decorators import login_required
 from django.shortcuts import get_object_or_404
 from django.urls import reverse_lazy
+from django.views.generic import DeleteView
 from django.shortcuts import render
 from django.http import HttpResponseRedirect
 from django.contrib.auth.decorators import login_required
 from django.views.generic.edit import FormView, CreateView, UpdateView
-from .models import Article
-from .forms import ArticleForm
+
+from .models import Article, Comment
+from .forms import ArticleForm, AddCommentForm
 
 
 def view(req, slug):
-    return render(req, "articles/detail.html")
+    article = get_object_or_404(Article, slug=slug)
+    return render(req, "articles/detail.html",
+                  {'article': article})
+
+
+@login_required()
+def add_comment(request, slug):
+    article = get_object_or_404(Article, slug=slug)
+
+    form = AddCommentForm()
+
+    if request.method == 'POST':
+        form = AddCommentForm(request.POST)
+        if form.is_valid():
+            new_comment = form.save(commit=False)
+            new_comment.body = form.cleaned_data['body']
+            new_comment.article = article
+            new_comment.author = request.user.profile
+            new_comment.save()
+            return redirect('article_view', slug=slug)
+
+    return render(request, 'articles/_add_comment.html', {'form': form, 'article': article})
```

Comment:
Dose this require the special redirect status codes?

Reply:
I'm using _top on the frame currently to get it to work.